src/components/victory-candlestick/victory-candlestick.js

```diff
@@ -0,0 +1,433 @@
+      const wickY1 = datum.high;
+      const wickY2 = datum.low;
+      const candleColor = datum.open > datum.close ?
+            props.candleColors.negative : props.candleColors.positive;
+      const candleWidth = 0.5 * (props.width - 2 * props.padding)/data.length;
+      const candleHeight = Math.max(datum.open, datum.close) - Math.min(datum.open, datum.close);
+      const candleY = Math.max(datum.open, datum.close);
+      const size = CandlestickHelpers.getSize(datum, props, calculatedProps);
+      const dataStyle = this.getDataStyles(datum, style.data);
+      const dataProps = defaults(
+        {},
+        this.getEventState(index, "data"),
+        props.dataComponent.props,
+        {
+          wickX, candleX, wickY1, wickY2, candleColor, candleWidth, candleHeight,
+          candleY, size, scale, datum, index, style: dataStyle, key: `point-${index}`
+        }
+      );
+      const candleComponent = React.cloneElement(props.dataComponent, assign(
+        {}, dataProps, {events: Helpers.getPartialEvents(dataEvents, index, dataProps)}
+      ));
+      const text = this.getLabelText(props, dataProps.datum, index);
+      if (text !== null && text !== undefined) {
+        const labelStyle = this.getLabelStyle(style.labels, dataProps);
+        const labelProps = defaults(
+          {},
+          this.getEventState(index, "labels"),
+          props.labelComponent.props,
+          {
+            key: `point-label-${index}`,
+            style: labelStyle,
+            x: x + labelStyle.padding * 2,
+            y: wickY1 + labelStyle.padding,
+            text,
+            index,
+            scale,
+            datum: dataProps.datum,
+            textAnchor: labelStyle.textAnchor,
+            verticalAnchor: labelStyle.verticalAnchor || "end",
+            angle: labelStyle.angle
+          }
+        );
+        const candleLabel = React.cloneElement(props.labelComponent, assign({
+          events: Helpers.getPartialEvents(labelEvents, index, labelProps)
+        }, labelProps));
+        return (
+          <g key={`point-group-${index}`}>
+            {candleComponent}
+            {candleLabel}
+          </g>
+        );
+      }
+      return candleComponent;
+    });
+  }
+
+  getCalculatedProps(props, style) {
+    const data = Data.getData(props);
+    // const range = {
+    //   x: Helpers.getRange(props, "x"),
+    //   y: Helpers.getRange(props, "y")
+    // };
+    // const domain = {
+    //   x: Domain.getDomain(props, "x"),
+    //   y: Domain.getDomain(props, "y")
+    // };
+    // const scale = {
+    //   x: Scale.getBaseScale(props, "x").domain(domain.x).range(range.x),
+    //   y: Scale.getBaseScale(props, "y").domain(domain.y).range(range.y)
+    // };
+    const scale = {
+      x: d3Scale.scaleLinear().domain([Math.min(data.map((datum) => {return datum.x;})), Math.max(data.map((datum) => {return datum.x;}))]).range([props.padding, props.width - props.padding]),
```

Comment:
We usually take the scale in as a prop that the user can determine, but in the case of a candle component with a very specific use, it might make sense to remove this option from the user. But we will need to think about how this will integrate with VictoryChart which expects to have control over the scale prop of all its children.